src/LibraryManager/Helpers/JsonHelpers.cs

```diff
@@ -1,15 +1,69 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using Microsoft.Web.LibraryManager.Contracts;
-using Microsoft.JSON.Core.Parser.TreeItems;
-using System.Linq;
 using System.Collections.Generic;
+using System.Linq;
+using Microsoft.JSON.Core.Parser;
+using Microsoft.JSON.Core.Parser.TreeItems;
+using Microsoft.Web.LibraryManager.Contracts;
 
-namespace Microsoft.Web.LibraryManager.Vsix
+namespace Microsoft.Web.LibraryManager.Helpers
 {
     internal static class JsonHelpers
     {
+        internal static JSONParseItem GetItemBeforePosition(int pos, JSONComplexItem parentItem)
+        {
+            JSONParseItem item = null;
+            JSONParseItemList children = parentItem.Children;
+            int start = 0;
+
+            if (children.Any())
+            {
+                start = children[0].Start;
+            }
+
+            if (start < pos)
+            {
+                int i = FindInsertIndex(children, pos, beforeExisting: true) - 1;
+
+                if (i >= 0)
+                {
+                    item = children[i];
+
+                    if (item is JSONComplexItem complexItem)
+                    {
+                        // Recurse to find the deepest item
+                        item = GetItemBeforePosition(pos, complexItem);
+                    }
+                }
+            }
+
+            return item;
+        }
+
+        internal static int FindInsertIndex(JSONParseItemList jsonParseItems, int rangeStart, bool beforeExisting)
```

Comment:
If this always going to be passed in as true, this flag can be removed.

Reply:
I think it can be removed.

---
In reply to: [203551539](https://github.com/aspnet/LibraryManager/pull/324#discussion_r203551539) [](ancestors = 203551539)